controller/Reporting.php

```diff
@@ -208,4 +217,47 @@ public function reports(){
         $requestOptions = $this->getRequestOptions();
         $this->returnJson(TestCenterHelper::getReports($testCenter, $requestOptions));
     }
+
+    /**
+     * Checks that delivery is finished and can be generated report
+     */
+    public function hasReport()
+    {
+
+        if ($this->hasRequestParameter("id")) {
+            $id = $this->getRequestParameter('id');
+        } else {
+            throw new \common_exception_MissingParameter('id', __METHOD__);
+        }
+
+        $idList = $id;
+        if (!is_array($idList)) {
+            $idList = [$idList];
+        }
+
+        /** @var $assessmentResultsService \oat\taoProctoring\model\AssessmentResultsService */
+        $assessmentResultsService = $this->getServiceManager()->get('taoProctoring/AssessmentResults');
+
+        $excepted = [];
```

Comment:
please use a more meaningful variable names so the next developer understand the meaning of your code (this comment is also applicable to other similar variables and file names: excepted, allExcepted, reportExceptedDeliveries, reportExceptedDeliveries.tpl, excepted etc.)